app/services/publishers/vacancy_sort.rb

```diff
@@ -0,0 +1,69 @@
+class Publishers::VacancySort < RecordSort
+  VALID_VACANCY_TYPES = %w[published pending draft expired awaiting_feedback].freeze
+
+  def initialize(organisation, vacancy_type)
+    @organisation = organisation
+    @vacancy_type = VALID_VACANCY_TYPES.include?(vacancy_type) ? vacancy_type : "published"
+    @column = send("#{@vacancy_type}_options").first.column
+    @order = send("#{@vacancy_type}_options").first.order
+  end
+
+  def options
+    send("#{@vacancy_type}_options").tap do |options|
```

Comment:
Not sure how I feel about this in terms of readability – I tend to restrict metaprogramming to libraries, or as a last resort. I reckon something like this might be more faffy but clearer in terms of intent:

```ruby
def options
  base_options.tap do |options|
    # ...
  end
end

def base_options
  case @vacancy_type
  when "pending"
    [
      SortOption.new("publish_on", "desc", I18n.t("jobs.sort_by.published_date.descending")),
      SortOption.new("expires_on", "asc", I18n.t("jobs.sort_by.expires_on.ascending")),
      SortOption.new("job_title", "asc", I18n.t("jobs.sort_by.job_title.ascending")),
    ]
  when "draft"
  # ...
  end
end
```